app/models/rule.rb

```diff
@@ -69,16 +70,20 @@ def canonical?
     (profiles & Profile.canonical).any?
   end
 
-  def self.from_openscap_parser(op_rule, benchmark_id: nil)
+  def self.from_openscap_parser(op_rule, benchmark_id: nil, group_id: nil)
     rule = find_or_initialize_by(ref_id: op_rule.id,
-                                 benchmark_id: benchmark_id)
+                                 benchmark_id: benchmark_id,
+                                 group_id: group_id)
 
     rule.op_source = op_rule
 
     rule.assign_attributes(title: op_rule.title,
                            description: op_rule.description,
                            rationale: op_rule.rationale,
                            severity: op_rule.severity,
+                           requires: op_rule.requires,
+                           conflicts: op_rule.conflicts,
```

Comment:
Can you please elaborate on the use of `requires` and `conflicts`? What it is for and what it should hold?

Reply:
Yeah, here's the description in the specification for xccdf:
<img width="639" alt="Screen Shot 2022-02-03 at 7 44 03 AM" src="https://user-images.githubusercontent.com/87209745/152354480-3528a19a-e670-48c9-9037-5590e0799a26.png">

So `requires` or `conflicts` can show up as an element under a Group or a Rule. `conflicts` will only have one `idref` of a certain Rule or Group that needs to be **unselected** because it conflicts with the current Group/Rule. `requires` can be a list of one or more ids that need to be **selected** in order for the current Group/Rule to be evaluated correctly. 

`requires` will show up like idref="A B C" when there's multiple ids so I split these into an array in the openscap parser but I was thinking it might be better to store them as a String in the database and split the string into an array on our end, in `compliance-backend`

Reply:
Considering that only two types can be linked (a Rule or a Group), would it make sense to create many-to-many relationships like `requires_rules` and `requires_groups` (retrospectively for conflicts as well) instead of the array within a string? I'd love to hear your thoughts.
/cc @skateman

Reply:
I like it :+1: just please don't forget that one of them is an array and the other one isn't

Reply:
Yep, this sounds like a good idea to me too. I'll make this update, thanks!